package/MDAnalysis/analysis/waterdynamics.py

```diff
@@ -1194,99 +1194,108 @@ class SurvivalProbability(object):
+        self.verbose = verbose
 
+        # backward compatibility
+        self.start = self.stop = self.tau_max = None
+        if t0 is not None:
+            self.start = t0
+            warnings.warn("t0 is deprecated, use run(start=t0) instead", category=DeprecationWarning)
 
-    def run(self):
-        """Analyze trajectory and produce timeseries"""
+        if tf is not None:
+            self.stop = tf
+            warnings.warn("tf is deprecated, use run(stop=tf) instead", category=DeprecationWarning)
 
-        # select all frames to an array
-        selected = self._selection_serial(self.universe, self.selection)
+        if dtmax is not None:
+            self.tau_max = dtmax
+            warnings.warn("dtmax is deprecated, use run(tau_max=dtmax) instead", category=DeprecationWarning)
 
-        if len(selected) < self.dtmax:
-            print ("ERROR: Cannot select fewer frames than dtmax")
-            return
+    def print(self, verbose, *args):
+        if self.verbose:
+            print(args)
+        elif verbose:
+            print(args)
 
-        for window_size in list(range(1, self.dtmax + 1)):
-            output = self._getMeanOnePoint(selected, window_size)
-            self.timeseries.append(output)
+    def run(self, tau_max=20, start=0, stop=-1, step=1, verbose=False):
+        """
+        Computes and returns the survival probability timeseries
+
+        Parameters
+        ----------
+        start : int
+            Zero-based index of the first frame to be analysed
+        stop : int
+            Zero-based index of the last frame to be analysed (inclusive)
+        step : int
+            Jump every `step`'th frame
+        tau_max : int
+            Survival probability is calculated for the range :math:`1 <= \tau <= tau_max`
+        verbose : Boolean
+            Overwrite the constructor's verbosity
+
+        Returns
+        -------
+        tau_timeseries : list
+            tau from 1 to tau_max
+        sp_timeseries : list
+            survival probability for each value of `tau`
+        """
 
+        # backward compatibility (and priority)
+        start = self.start if self.start is not None else start
+        stop = self.stop if self.stop is not None else stop
+        tau_max = self.tau_max if self.tau_max is not None else tau_max
 
-    def _selection_serial(self, universe, selection_str):
-        selected = []
-        pm = ProgressMeter(self.tf-self.t0, interval=10,
-                           verbose=True, offset=-self.t0)
-        for ts in universe.trajectory[self.t0:self.tf]:
-            selected.append(universe.select_atoms(selection_str))
-            pm.echo(ts.frame)
-        return selected
+        # sanity checks
+        if stop >= len(self.universe.trajectory):
+            raise ValueError("\"stop\" must be smaller than the number of frames in the trajectory.")
 
+        if stop == -1:
+            stop = self.universe.trajectory[-1].frame + 1
```

Comment:
this will load the frame, aka slow.  Can you just use `None` for stop?  Or maybe `len(u.trajectory)` when you need to check bounds?

Reply:
Sounds much better. Done